rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/rmw_context_impl.hpp

```diff
@@ -17,7 +17,7 @@
 
 #include <mutex>
 
-struct rmw_context_impl_t
+struct rmw_context_impl_s
```

Comment:
Ah, I see what is happening here.

Over in https://github.com/ros2/rmw/pull/313/files#diff-0460a36c52d1df8664f7ab0ee834d07ac10cfaf17fda75f7477421e4aadf30d8R34, we are doing `typedef struct rmw_context_impl_s rmw_context_impl_t`, which we are then using later on.  So here in the RMW implementation, we need to define this as `struct rmw_context_impl_s`, but *not* make it `struct rmw_context_impl_t` (since that is done at the `rmw` layer, not the implementation layer).

I guess we do this in all of the RMW implementations, so this makes sense.  It also means that, in theory, we could write a RMW layer purely in C (though I doubt we would ever do that).

This change is OK with me, but this is definitely going to have repercussions for RMW implementations that are not in the default build.  We'll want to put a release note in place for this.  I also think we should probably put a comment here explaining that this is the implementation for the stuff over in `rmw/init.h`, since understanding this is pretty tricky.

Reply:
See https://github.com/ros2/rmw_fastrtps/pull/558/commits/e2b22c7130a4d67188399a2166f185eefb37d688.